### PR TITLE
KFSPTS-21717 Initial XML converter updates for KEW

### PR DESCRIPTION
**Pull request by cah292**
This PR represents the first phase of remediating our maintenance XML conversion feature for KEW-to-KFS compatibility. For this initial effort, the copied-over Cynergy classes have been consolidated into the cu-kfs-specific variants, and one of the utility classes has been simplified to only contain the methods that are truly needed. In addition, the XML conversion rules file has been updated to include the relevant basic rule changes for the KEW-to-KFS upgrade, and has also been updated to include a modified version of the FINP-7352 rule changes from the 2021-02-18 financials patch.

The recent KFS Parameter maintenance documents can now be opened successfully when running the upgraded cu-kfs code locally. More conversion rule changes are needed to allow for properly opening Parameter documents migrated from Cynergy, but that work will be deferred to the KFSPTS-21720 user story.

Also, a new Parameter document unit test case has been added, and some quick cleanup has been done on the unit test class to make it use JUnit 5.

### CODE REVIEW

**Comment by jhulslander** [comment]
retest this please

**Comment by jhulslander** [comment]
Please note, there are test failures, for some reason  the results aren't being sent back to github

**Comment by jhulslander** [comment]
retest this please

**Comment by jhulslander** [comment]
@cah292 ,
Please check out the errors on the unit test run.  I think they are real now.  I will need to look into why Jenkins isn't returning an error state to github.

File: src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml [hunk 1]
```diff
@@ -501,38 +448,78 @@
...
+                <replacement>org.kuali.kfs.coreservice.impl.parameter.Parameter</replacement>
+            </pattern>
+            <pattern>
+                <match>org.kuali.kfs.coreservice.impl.parameter.ParameterBo</match>
+                <replacement>org.kuali.kfs.coreservice.impl.parameter.Parameter</replacement>
             </pattern>
             <pattern>
                 <match>org.kuali.rice.kew.rule.RuleDelegation</match>
-                <replacement>org.kuali.rice.kew.rule.RuleDelegationBo</replacement>
+                <replacement/>
```
Comment (db483) [inline src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml:474]: Is this replacement correct?

File: src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml [hunk 2]
```diff
@@ -501,38 +448,78 @@
...
+            </pattern>
+            <pattern>
+                <match>org.kuali.rice.kew.rule.RuleDelegationBo</match>
+                <replacement/>
```
Comment (db483) [inline src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml:478]: Is this replacement correct?

**Review by db483 — CHANGES_REQUESTED** [review]
Changes look good but I have a couple of questions about some conversion rules. Thank you.

**Comment by jhulslander** [comment]
retest this please

File: src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml (hunk 1 above)
Comment (cah292) [reply src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml:474]: KFS does not appear to have its own equivalent of Rice's RuleDelegation or RuleDelegationBo objects. Thus, this rule and the one below will remove them from the maintenance XML if they're somehow present.

The base financials upgrade rules were updated to just remove and RuleDelegation or RuleDelegationBo mappings altogether. If you would prefer, I could remove those mappings from our rules file as well.

File: src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml (hunk 2 above)
Comment (cah292) [reply src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml:478]: See my comment on the RuleDelegation pattern above.

File: src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml (hunk 1 above)
Comment (db483) [reply src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml:474]: Thank you for explaining this Chad. What maintenance xml do we expect could contain these 2?

Comment (cah292) [reply src/main/resources/edu/cornell/kfs/krad/config/MaintainableXMLUpgradeRules.xml:474]: Thanks for the input. I'll get those unneeded rules removed.

**Comment by cah292** [comment]
I've removed the unneeded RuleDelegation-related conversion rules. This is ready for review again.

**Review by db483 — APPROVED** [review]
Changes look good. Unit test are currently failing but the failures are not related to the changes on this PR. Any issues with the unit tests will be resolved on a separate user story.

**Pull request merged**